bigchaindb/util.py

```diff
@@ -251,16 +373,87 @@ def verify_signature(signed_transaction):
         bool: True if the signature is correct, False otherwise.
     """
 
-    data = signed_transaction.copy()
+    for fulfillment in signed_transaction['transaction']['fulfillments']:
+        fulfillment_message = get_fulfillment_message(signed_transaction, fulfillment)
+        # verify the fulfillment (for now lets assume there is only one owner)
+        try:
+            parsed_fulfillment = cc.Fulfillment.from_uri(fulfillment['fulfillment'])
+        except Exception:
+            return False
+        is_valid = parsed_fulfillment.validate(serialize(fulfillment_message))
+
+        # if not a `CREATE` transaction
+        if fulfillment['input']:
+            is_valid &= parsed_fulfillment.condition.serialize_uri() == \
+                fulfillment_message['condition']['condition']['uri']
+        if not is_valid:
+            return False
+
+    return True
+
+
+def get_fulfillment_message(transaction, fulfillment, serialized=False):
+    """Get the fulfillment message for signing a specific fulfillment in a transaction
+
+    Args:
+        transaction (dict): a transaction
+        fulfillment (dict): a specific fulfillment (for a condition index) within the transaction
+        serialized (bool): False returns a dict, True returns a serialized string
```

Comment:
Not sure I understand. The flag is just to indicate if you want to return the result serialized or not.
We can remove that option and force however uses this method to always serialize afterwards if needed